[release] 1.9.0.Alpha2 announcement (#778)
debezium/debezium.github.io: _posts/2022-02-09-debezium-1-9-alpha2-released.adoc

---
layout: post
title:  Debezium 1.9.0.Alpha2 Released
date:   2022-02-09
tags: [ releases, mysql, postgres, sqlserver, cassandra, oracle, db2, vitess, outbox ]
author: anmohant
---

It's my pleasure to announce the second release of the Debezium 1.9 series, *1.9.0.Alpha2*!

This release includes support for Oracle 21c,

> improvements around Redis for Debezium Server, configuring `KAFKA_QUERY_TIMEOUT`,

Reviewer: ```suggestion
improvements around Redis for Debezium Server, configuring the `kafka.query.timeout.ms` option,
```

and a number of bug fixes around DDL parsers, build infrastructure, etc.

Overall, the community fixed https://issues.redhat.com/issues/?jql=project%20%3D%20DBZ%20AND%20fixVersion%20%3D%201.9.0.Alpha2%20ORDER%20BY%20issuetype%20DESC[51 issues] for this release. Let’s take a closer look at some of the highlights.

+++<!-- more -->+++

== Support for Oracle 21c

> The Debezium Oracle connector now supports the Oracle 21.3.0 with tested LogMiner and Xstreams compatibility.

Reviewer: ```suggestion
The Debezium Oracle connector has been tested with the latest release of Oracle 21c, 21.3.0.0, and is compatible.  If you use either the LogMiner or the Xstreams adapter, you should now be able to use Oracle's latest flagship version and stream change events without any changes.  If you are on Oracle 12 or Oracle 19 and perform a database upgrade, your connector configuration should require no changes and remain compatible.
```

> == Configuring `KAFKA_QUERY_TIMEOUT`

Reviewer: ```suggestion
== Configuring `kafka.query.timeout.ms`
```

When making API calls using Kafka Admin Client the default timeout was set to 3 seconds.
But having this smaller timeout was insufficient for large TLS or SSL encryption, where the requests got timeout before completion.
Now we have `KAFKA_QUERY_TIMEOUT` as a configurable field, it can be used to provide the wait time for fetching information from the Kafka Admin Client,

> while the default timeout remains 3 seconds.

Reviewer: ```suggestion
When using the Kafka Admin Client and issuing API calls, the default timeout is 3 seconds.
The new `kafka.query.timeout.ms` field can be used to provide a custom timeout to the Kafka Admin Client to avoid possible timeout problems in environments that may use TLS or SSL encryption or where network latency causes an unexpected timeout.
```

Thanks to the great work done by community member, https://github.com/snigdhasjg[Snigdhajyoti Ghosh].

== Improvements in Redis for Debezium Server

We have three new fields in the Redis support for Debezium Server

* `redis.retry.initial.delay.ms`
* `redis.retry.max.delay.ms`
* `batch.size`

The `maxmemory` configuration in Redis allows to provide a memory limit.
If `maxmemory` is not configured, Redis keeps allocating memory which gradually eats up all free memory.
This results in OOM errors and resolves as soon as entries are removed from the stream.

> `redis.retry.initial.delay.ms` and `redis.retry.max.delay.ms` is used in setting initial and max retry delays when encountering Redis connection or OOM issues.

Reviewer: ```suggestion
Redis allows specifying a maximum memory limit using the `maxmemory` configuration; however, if this field is not configured then Redis will continue to allocate memory.
If all memory is consumed, an OutOfMemory exception occurs.
The Redis sink now uses `redis.retry.initial.delay.ms` and `redis.retry.max.delay.ms` to set an initial and max-retry delay to be more resilient to this and connection-related issues.  If you have or are experiencing such exceptions, we urge you to try these new settings to improve the sink's resilience and experience.
```

> Using pipeline transactions helps in speeding up Redis queries.

Reviewer: ```suggestion
Pipeline-based transactions can substantially increase Redis queries.
```

> `batch.size` is used to provide the number of change records to insert in a single batch write.

Reviewer: ```suggestion
In order to leverage pipeline-based transactions, the `batch.size` configuration option can be specified which will allow Redis to write batches of change records rather than each record one by one.
```
